372 compendium scene

SkyRiser568 opened the pull request:
Added monster compendium scene that shows detailed information for a monster when hovered over, designers will be able to edit what information is displayed for each monster.

📝 Checklist before requesting a review:

- [x]  I have run this branch locally
- [x]  I have performed a self-review of my code
- [x]  I have confirmed critical features still function

CrispyKorn commented on MonstieWash/Assets/Scripts/Compendium/CompendiumEntry.cs:
```diff
@@ -0,0 +1,34 @@
+
+public class CompendiumEntry : MonoBehaviour
+{
+    public string MonsterName { get => monsterName;}
+    public string Temperament { get => temperament; }
+    public string Description { get => description;}
+    public Sprite ImageOriginal { get => imageOriginal; }
+    public Sprite ImageCleared { get => imageCleared; }
+    public bool Completed { get => completed;}
+
+    [SerializeField] private string monsterName;
+    [SerializeField] [TextAreaAttribute] private string temperament;
+    [SerializeField] [TextAreaAttribute] private string description;
+    [SerializeField] private Sprite imageOriginal;
+    [SerializeField] private Sprite imageCleared;
+    [SerializeField] private bool completed;
+
+    private CompendiumManager m_compendiumManager;
+
```
Just to keep things consistent with the layout of other scripts, the public properties will need to move to the bottom of the declaration section.

```suggestion
    [SerializeField] private string monsterName;
    [SerializeField] [TextAreaAttribute] private string temperament;
    [SerializeField] [TextAreaAttribute] private string description;
    [SerializeField] private Sprite imageOriginal;
    [SerializeField] private Sprite imageCleared;
    [SerializeField] private bool completed;

    private CompendiumManager m_compendiumManager;

    public string MonsterName { get => monsterName;}
    public string Temperament { get => temperament; }
    public string Description { get => description;}
    public Sprite ImageOriginal { get => imageOriginal; }
    public Sprite ImageCleared { get => imageCleared; }
    public bool Completed { get => completed;}
```

CrispyKorn commented on MonstieWash/Assets/Scripts/Compendium/CompendiumManager.cs:
```diff
@@ -0,0 +1,43 @@
+        }
+    }
+
+    public void DisplayEntry(CompendiumEntry entry)
```
I know its a pretty simple function but just be sure to add XML for it anyways.

CrispyKorn commented on MonstieWash/Assets/Scripts/Compendium/CompendiumEntry.cs:
```diff
@@ -0,0 +1,34 @@
+        m_compendiumManager = FindFirstObjectByType<CompendiumManager>();
+    }
+
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+        m_compendiumManager.DisplayEntry(this);
+    } 
+}
```
This is probably the most significant issue, in that any collider will trigger this function. This means that the arm of the player triggers it as well as the interaction box of the hand, causing some awkward and unintuitive behaviour. 

We want this function to run only when the hand box collider is overlapping. To achieve this, either only check for box colliders as the hand collider is a box2d while the arm collider is a polygon2d, OR mess with the layer interactions of the colliders so that only the hand collides with the monster elements. Up to you how you think is best to handle this, but it'll need to be resolved before it can be merged.

CrispyKorn reviewed (changes requested):
Just fix up a couple things based on my comments and also please delete the old compendium scene to keep things clean.

Looks good though, well done!

SkyRiser568 commented:
Fixed the stuff you requested. With the hand collider, I ended up using IgnoreCollision to ignore the poly collider on the arm, also the hand itself had no rigidbody2d on it so it wasn't able to trigger any collisions so I added one set to kinematic, not sure if it will mess with anything but the game still worked fine when I tested it.

CrispyKorn reviewed (approved):
Nice, works great now!
